packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+        rekeyForeignKeys(_.omit(row, "id"), idLookupByTable)
+    );
+
+    const inserted = await knex(tableName).insert(rowsToInsert).returning("*");
+    const idLookup = _.chain(inserted)
+        .map((insertedRow, idx) => [rowsToInsert[idx].id, insertedRow.id])
+        .fromPairs()
+        .value();
+
+    return { inserted, idLookup };
+}
+
+function getAllTenantIds(dbContents) {
+    return _.chain(dbContents)
+        .values()
+        .flatten()
+        .map("tenant_id")
+        .uniq()
+        .filter((x) => x !== undefined)
+        .sort()
+        .value();
+}
+
+async function importTenants(dbContents, idLookupByTable) {
+    // First, we ask users for tenant names and main agencies for all tenants that
+    // will be created (this should be just one, but technically could be more than
+    // one, so we have to account for it here)
+    const allTenantIds = getAllTenantIds(dbContents);
+    const agencyOptionsByTenantId = _.fromPairs(
+        allTenantIds.map((tenantId) => [
+            tenantId,
+            dbContents.agencies.filter(
+                (agency) => agency.tenant_id === tenantId
+            ),
+        ])
+    );
+    const questions = _.chain(allTenantIds)
+        .map((tenantId) => [
+            {
+                type: "input",
+                name: `tenantNames.${tenantId}`,
+                message: `Tenant name for source tenant ${tenantId}`,
+                validate: (name) =>
+                    name.trim().length > 0 || "must enter a name",
+            },
+            {
+                type: "search-list",
+                name: `mainAgencies.${tenantId}`,
+                message: `Main agency for source tenant ${tenantId}`,
+                choices: [
+                    {
+                        name: "(create new agency with same name as tenant)",
+                        value: null,
+                    },
+                    ...agencyOptionsByTenantId[tenantId].map((agency) => ({
+                        name: agency.name,
+                        value: agency.id,
+                    })),
+                ],
+            },
+        ])
+        .flatten()
+        .value();
+    let { tenantNames, mainAgencies } = await inquirer.prompt(questions);
+
+    // Annoying: by default Inquirer will make (sparse) arrays instead of objects; convert them back
+    // to the expected form.
+    tenantNames = _.fromPairs(
+        tenantNames
+            .filter((x) => x !== undefined)
+            .map((name, idx) => [idx, name])
+    );
+    mainAgencies = _.fromPairs(
+        mainAgencies
+            .filter((x) => x !== undefined)
+            .map((agencyId, idx) => [idx, agencyId])
+    );
+
+    // Create tenants
+    let inserted = [];
+    for (const tenantId of allTenantIds) {
+        const tenantName = tenantNames[tenantId];
+        const mainAgencyId = mainAgencies[tenantId];
+
+        const tenant = await knex("tenants")
+            .insert({
```

Comment:
Good idea. I think probably the easiest way to do that would be to put all of the DB inserts in a transaction and make it so that the file copy portion of the script can be run separately.